### PR TITLE
Include spconv header files in the sdist package

### LINKED ISSUE #1743: CreateProcess failed: The system cannot find the file specified. returned while compiling on Windows
Thanks for reporting the unexpected results and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmcv.readthedocs.io/en/latest/trouble_shooting.html) but cannot get the expected help.
3. The unexpected results still exist in the latest version.

**Describe the Issue**
A clear and concise description of what the bug is, including what results are expected and what the real results you got.

**Reproduction**

1. What command, code, or script did you run?

```bash
A placeholder for the command.
```

2. Did you make any modifications on the code? Did you understand what you have modified?

**Environment**

1. Please run `python -c "from mmcv.utils import collect_env; print(collect_env())"` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
```
10:56:02  FAILED: D:/Workspace/Jenkins/workspace/mmcv/build-on-windows-commit@2/mmcv/build/temp.win-amd64-3.7/Release/mmcv/ops/csrc/pytorch/deform_conv.obj 
10:56:02  cl /showIncludes /nologo /Ox /W3 /GL /DNDEBUG /MD /MD /wd4819 /EHsc -DMMCV_WITH_CUDA -ID:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\common -ID:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\common\cuda -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\torch\csrc\api\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\TH -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\THC "-ID:\NVIDIA GPU Computing Toolkit\CUDA\v10.2\include" -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" -c D:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\pytorch\deform_conv.cpp /FoD:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\build\temp.win-amd64-3.7\Release\mmcv/ops/csrc/pytorch\deform_conv.obj -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 /std:c++14
10:56:02  CreateProcess failed: The system cannot find the file specified.
10:56:02  ninja: build stopped: subcommand failed.
10:56:02  Traceback (most recent call last):
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1400, in _run_ninja_build
10:56:02      check=True)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\subprocess.py", line 512, in run
10:56:02      output=stdout, stderr=stderr)
10:56:02  subprocess.CalledProcessError: Command '['ninja', '-v', '-j', '8']' returned non-zero exit status 1.
10:56:02  
10:56:02  During handling of the above exception, another exception occurred:
10:56:02  
10:56:02  Traceback (most recent call last):
10:56:02    File "setup.py", line 391, in <module>
10:56:02      zip_safe=False)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\__init__.py", line 153, in setup
10:56:02      return distutils.core.setup(**attrs)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\core.py", line 148, in setup
10:56:02      dist.run_commands()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\dist.py", line 966, in run_commands
10:56:02      self.run_command(cmd)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\dist.py", line 985, in run_command
10:56:02      cmd_obj.run()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\command\build_ext.py", line 79, in run
10:56:02      _build_ext.run(self)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 340, in run
10:56:02      self.build_extensions()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 580, in build_extensions
10:56:02      build_ext.build_extensions(self)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 449, in build_extensions
10:56:02      self._build_extensions_serial()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 474, in _build_extensions_serial
10:56:02      self.build_extension(ext)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\command\build_ext.py", line 202, in build_extension
10:56:02      _build_ext.build_extension(self, ext)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 534, in build_extension
10:56:02      depends=ext.depends)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 562, in win_wrap_ninja_compile
10:56:02      with_cuda=with_cuda)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1140, in _write_ninja_file_and_compile_objects
10:56:02      error_prefix='Error compiling objects for extension')
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1413, in _run_ninja_build
10:56:02      raise RuntimeError(message)
10:56:02  RuntimeError: Error compiling objects for extension
10:56:04  Start python setup.py build_ext.
10:56:04  Python setup.py build_ext failed.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

### LINKED ISSUE #1752: “utils/spconv/spconv/geometry.h”: No such file or directory
I use this command to install mmcv-full
`pip install mmcv-full -f https://download.openmmlab.com/mmcv/dist/cu102/torch1.9.0/index.html`
and find this problem
` ./mmcv/ops/csrc/pytorch/cpu\sparse_indice.cpp(16): fatal error C1083: 无法打开包括文件: “utils/spconv/spconv/geometry.h”: No such file or directory`
` error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Professional\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2`
When I install in win10 and Linux both meet this problem.
Thanks for replying!

### LINKED ISSUE #1762: Missing maxpool.h while install mmcv-full
When I install mmcv-full，I got a error that shows missing the maxpool.h file.
```
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  ./mmcv/ops/csrc/pytorch/cpu/sparse_maxpool.cpp:16:10: fatal error: utils/spconv/spconv/maxpool.h: 没有那个文件或目录
   #include <utils/spconv/spconv/maxpool.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for mmcv-full
```

The cuda version of my computer is 11.3
and torch version is 1.10.2
How to solve the problem?
Thank you for your help

### PR DESCRIPTION
**Pull request by wangruohui**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Try to fix #1752, fix #1762, and may fix #1743.

## Modification

Include missing header files in MANIFEST.in 

## BC-breaking (Optional)

No I think. 

## Use cases (Optional)

When people want to install (with compilation) mmcv-full using `pip install mmcv-full` directly from pypi instead of from source code. 

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by wangruohui** [comment]
Need more tests. 

An idea is to build an sdist package from the repo using `python setup.py sdist`. Then copy the resulted `dist/mmcv-full-1.4.6.tar.gz` (which should contain missing headers) to some temporary place to unpack. Then `cd` into the temporary folder and try `python setup.py build_ext`. 

But I don't have a powerful computer at hand now.

**Comment by grimoire** [comment]
> Need more tests.
> 
> An idea is to build an sdist package from the repo using `MMCV_WITH_OPS=1 python setup.py sdist`. Then copy the resulted `dist/mmcv-full-1.4.6.tar.gz` (which should contain missing headers) to some temporary place to unpack. Then `cd` into the temporary folder and try `python setup.py build_ext`.
> 
> I have successfully compiled on my mac and pass `test/test_ops` despite a lot of warnings..

failed with log:
```
fatal error: ../spconv_utils.h: No such file or directory
```

**Comment by wangruohui** [comment]
> > Need more tests.
> > An idea is to build an sdist package from the repo using `MMCV_WITH_OPS=1 python setup.py sdist`. Then copy the resulted `dist/mmcv-full-1.4.6.tar.gz` (which should contain missing headers) to some temporary place to unpack. Then `cd` into the temporary folder and try `python setup.py build_ext`.
> > I have successfully compiled on my mac and pass `test/test_ops` despite a lot of warnings..
> 
> failed with log:
> 
> ```
> fatal error: ../spconv_utils.h: No such file or directory
> ```

Fixed. 
I tried on another windows computer with CUDA and pass the test. 

Moreover, I think we can modify the logic of CI to simulate such installation process.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -111,8 +111,16 @@ jobs:
       # pstuil is an optional package to detect the number of CPU for compiling mmcv
       - name: Install psutil
         run: pip install psutil
-      - name: Build and install
-        run: rm -rf .eggs && pip install -e .
+      - name: Create sdist and untar
+        run: |
+          MMCV_WITH_OPS=1 python setup.py sdist
+          tar zxvf dist/mmcv-full* -C /tmp
```
Comment (ZwwWayne) [inline .github/workflows/build.yml:117]: we do not need to untar the package, we can simply use `pip install dist/mmcv-full* `

**Review by grimoire — APPROVED** [review]
LGTM

**Pull request merged**